WordPress/gutenberg
# Adds button component tests.

Adds basic button component tests.  Covers cases where the tag name or
other properties are conditionally rendered.  Related to progress on
#641.  This PR adds the components directory to the list of entry points
configured for tests by webpack.

This introduces Enzyme and the react-test-renderer library as dev dependencies.

**Testing Instructions**
Run `npm i && npm run test-unit` ensure tests pass.  Change Button logic to ensure tests fail as they should.
